src/services/TestTypesService.ts

```diff
@@ -197,6 +207,50 @@ export class TestTypesService {
                     bool = testCode.forVehicleAxles === filterExpression.vehicleAxles;
                 }
                 break;
+            case "forEuVehicleCategory":
+                if (Array.isArray(testCode.forEuVehicleCategory)) {
```

Comment:
can simplify all of these as a single function:
const filterOnField = (filterVal) => {
                if (Array.isArray(testCode[filterVal])) {
                    testCode[filterVal].map((arrayElement) => {
                        if (arrayElement === filterExpression[getFilterFieldWithoutFor(filterVal)]) {
                            bool = true;
                        }
                    });
                } else {
                    bool = testCode[filterVal]=== filterExpression[getFilterFieldWithoutFor(filterVal)];
                }
      }

with

const getFilterFieldWithoutFor = (filterVal) => {
   const rightLetters = filterVal.slice(3); //cut off the leading "for", but still got a capital letter leading
   return rightLetters[0].toLowerCase() + rightLetters.slice(1) // switch first letter to lower case, and rejoin with rest of string
}

Would let you remove the case statement entirely, and replace it with just a filterOnField line

Reply:
As discussed, not my favourite solution but does remove code duplication. Done.